Sources/DarkModeCore/DMTraitCollection.h

```diff
@@ -15,14 +17,21 @@ typedef NS_ENUM(NSInteger, DMUserInterfaceStyle) {
 
 @interface DMTraitCollection : NSObject
 
-@property (class, nonatomic, strong) DMTraitCollection *currentTraitCollection;
+@property (class, nonatomic, readonly) DMTraitCollection *currentTraitCollection;
+@property (class, nonatomic, readonly) DMTraitCollection *lastManuallySetTraitCollection;
 
 + (DMTraitCollection *)traitCollectionWithUserInterfaceStyle:(DMUserInterfaceStyle)userInterfaceStyle;
++ (DMTraitCollection *)traitCollectionWithUITraitCollection:(UITraitCollection *)traitCollection API_AVAILABLE(ios(13.0));
 
 @property (nonatomic, readonly) DMUserInterfaceStyle userInterfaceStyle;
+@property (nonatomic, readonly) UITraitCollection *uiTraitCollection API_AVAILABLE(ios(13.0));
 
 - (instancetype)init NS_DESIGNATED_INITIALIZER;
 
++ (void)registerApplication:(nullable UIApplication *)application;
```

Comment:
Maybe we should provide more APIs to sync user interface style. For example, there is no `UIApplication` instance in app extensions, instead `UIViewController` instance can also be used to override user interface style. I think we can provide these APIs:

```swift
+ (void)syncWithApplication:(UIApplication *)application; // convenience method for + (void)syncWithViews:(NSArray<UIView *> *)views; since UIWindow is subclass of UIView
+ (void)syncWithViewController:(UIViewController *)viewController;
+ (void)syncWithViews:(NSArray<UIView *> *)views;
+ (void)stopSync;
```

To avoid adding multiple properties, I think we can use a single block property.

Thoughts? @eyvallah @kylinroc

Reply:
I agree, for app extension, one has to resort to UIViewController, so it is not made possible with this PR. how do you implement using a single block property though?

Reply:
For reference:

```swift
private var modeChangeHandler: (() -> Void)?

public func sync(with application: UIApplication) {
    sync(with: application.windows)
}

public func sync(with views: [UIView]) {
    modeChangeHandler = { [weak self] in
      // views should be weak reference, probably implement with NSPointerArray
      guard let self = self else { return }

      for view in views {
        view.overrideUserInterfaceStyle = self.userInterfaceStyle
      }
    }

    modeChangeHandler!()
}
```

Reply:
I added more parameters in the last commit, so the API look like these. I still prefer register here, since syncing (calling the block after setting the value) does not have to be mandatory

+ (void)registerWithApplication:(UIApplication *)application syncImmediately:(BOOL)syncImmediately animated:(BOOL)animated;
+ (void)registerWithViewController:(UIViewController *)viewController syncImmediately:(BOOL)syncImmediately animated:(BOOL)animated;
+ (void)unregister;